### PR TITLE
[SIG-4001] Error message component already in Dom to help screen readers

### PR DESCRIPTION
**Pull request by kiruys**
FileUpload, FileInput, textarea, and FormField component

### CODE REVIEW

File: src/components/ErrorMessage/ErrorMessage.tsx [hunk 1]
```diff
@@ -10,8 +10,15 @@ import {
...
-}))`
+}))<{ invalid: boolean }>`
   margin-top: 0;
+  white-space: pre-wrap;
+  display: none;
+  ${({ invalid }) =>
+    invalid &&
+    css`
+      display: flex;
+    `}
```
Comment (janjaap) [inline src/components/ErrorMessage/ErrorMessage.tsx:21]: This change won't solve the issue at hand; the point is that the element that holds the error message, should be present in the DOM. Setting its `display` value to 'none' will effectively hide it for browsers and screen readers which defeats the purpose.
See https://www.w3.org/WAI/WCAG21/working-examples/aria-role-status-searchresults/ for an example.

**Review by janjaap — CHANGES_REQUESTED** [review]

File: src/components/FileInput/FileInput.tsx [hunk 2]
```diff
@@ -84,14 +84,11 @@ const FileInput: FunctionComponent<FileInputProps> = ({
...
-          <StyledErrorMessage
-            key={index}
-            data-testid="fileInputError"
-            message={message}
-          />
-        ))}
+      <StyledErrorMessage
+        data-testid="fileInputError"
+        message={errorMessages.join('\n') || ''}
+        invalid={errorMessages.length > 0}
```
[the remark this answers is not in the document]
Comment (kiruys) [reply src/components/FileInput/FileInput.tsx:90]: invalid props is removed

File: src/signals/incident/components/form/FormField/FormField.tsx [hunk 3]
```diff
@@ -81,59 +81,60 @@ const FormField: FunctionComponent<FormFieldProps> = ({
...
+                      .requiredLength
+                  )} tekens ingevoerd`
+                : ''
+            }
+          />
+
+          <ErrorMessage
+            data-testid="customError"
+            message={hasError('custom') ? getError('custom') : ''}
+          />
```
Comment (janjaap) [inline src/signals/incident/components/form/FormField/FormField.tsx:136]: These only need to be rendered when there is an actual error to display.

File: src/components/TextArea/TextArea.tsx [hunk 4]
```diff
@@ -78,13 +78,13 @@ const TextArea = forwardRef<HTMLTextAreaElement, TextAreaProps>(
...
         )}
-
-        {errorMessage && (
+        <div role="status">
           <StyledErrorMessage
             id="textareaErrorMessage"
-            message={errorMessage}
+            data-testid="textareaError"
+            message={errorMessage || ''}
           />
```
Comment (janjaap) [inline src/components/TextArea/TextArea.tsx:86]: This component should only be rendered when there is an actual error message that needs to be shown.

**Review by janjaap — CHANGES_REQUESTED** [review]

**Comment by kiruys** [comment]
Created new PR to fix this

**Pull request closed without merging**